backend/tests/test_saves.py

```diff
@@ -0,0 +1,76 @@
+from saves import Saves
+import json, uuid
+
+
+def fake_game_round(round=1, gid='9697f0f5-2733-4a7e-98c4-6c1b8fb6d7e6'):
+
+    return {"round": round, "gameid": gid , 
+            "state": { "table": [], "deck": [], "players": [], "hands": {}, "capture": {}, "player_order": [] } }
+
+
+def test_save_single():
+    s = Saves(DBPath='test.db')
+    s._initdb()
+
+    gameid = str(uuid.uuid4())
+    fake_game_round_1 = fake_game_round(gid=gameid)
+    gameplays = [ fake_game_round_1 , ]
+    s.update(gameid, gameplays)    
+
+    g = s.get_id(gameid) 
+    assert g  
+    print(g['id'])
+    assert g['id']==gameid 
+
+
+
+def test_save_multiple_rounds():
+    s = Saves(DBPath='test.db')
```

Comment:
Pytest has fixtures that can help with creation/teardown of databases when testing:
https://docs.pytest.org/en/6.2.x/fixture.html
```python
def active_game():
    db = sqlite3.connect(':memory:')
    c = db.cursor()
    # If you wanna prepopulate things...
    c.execute('''CREATE...''')
    c.execute('INSERT INTO ...''')
    db.commit()
    yield c
    db.close()
```
Then when testing you can do:
```python
@pytest.mark.usefixtures("active_game")
def test_game_state(active_game):
  c = active_game
  ...
  ```

Reply:
thanks. that's much better approach. i couldn't get it to run right away though so removed this one until I understand fixture.